shared/bundle_analysis/parsers/v3.py

```diff
@@ -0,0 +1,424 @@
+        self.module_chunk_unique_external_ids_index = {}
+
+        # misc. top-level info from the stats data (i.e. bundler version, bundle time, etc.)
+        self.info = {}
+
+        # temporary parser state
+        self.session = None
+        self.asset = None
+        self.chunk = None
+        self.chunk_asset_names = []
+        self.module = None
+        self.module_chunk_unique_external_ids = []
+
+        self.asset_list = []
+        self.chunk_list = []
+        self.module_list = []
+
+        # dynamic imports: mapping between Chunk and each file name of its dynamic imports
+        self.dynamic_imports_mapping = defaultdict(
+            list
+        )  # typing: Dict[Chunk, List[str]]
+
+    @sentry_sdk.trace
+    def parse(self, path: str) -> Tuple[int, str]:
+        try:
+            self.reset()
+
+            # Retrieve the info section first before parsing all the other things
+            # this way when an error is raised we know which bundle plugin caused it
+            with open(path, "rb") as f:
+                for event in ijson.parse(f):
+                    self._parse_info(event)
+
+            self.session = Session(info={})
+            self.db_session.add(self.session)
+            self.db_session.flush()
+
+            with open(path, "rb") as f:
+                for event in ijson.parse(f):
+                    self._parse_event(event)
+
+                # Delete old session/asset/chunk/module with the same bundle name if applicable
+                old_session = (
+                    self.db_session.query(Session)
+                    .filter(
+                        Session.bundle == self.session.bundle,
+                        Session.id != self.session.id,
+                    )
+                    .one_or_none()
+                )
+                if old_session:
+                    for model in [Asset, Chunk, Module]:
+                        to_be_deleted = self.db_session.query(model).filter(
+                            model.session == old_session
+                        )
+                        for item in to_be_deleted:
+                            self.db_session.delete(item)
+                            self.db_session.flush()
+                    self.db_session.delete(old_session)
+                    self.db_session.flush()
+
+                if self.asset_list:
+                    insert_asset = Asset.__table__.insert().values(self.asset_list)
+                    self.db_session.execute(insert_asset)
+
+                # Needs to use ORM-style insert to update the models since they
+                # will be used later in dynamic import processing
+                self.db_session.add_all(self.chunk_list)
+
+                if self.module_list:
+                    insert_modules = Module.__table__.insert().values(self.module_list)
+                    self.db_session.execute(insert_modules)
+
+                self.db_session.flush()
+
```

Comment:
Below is where we have the list of items to insert into the DB. Also a small above change
```
self.db_session.add_all(self.chunk_list)
```
compared to before of `Chunk.__table__.insert().values`